crates/services/gas_price_service/src/fuel_gas_price_updater/da_source_adapter/service.rs

```diff
@@ -13,39 +11,93 @@ use std::{
     time::Duration,
 };
 use tokio::{
-    sync::mpsc::Sender,
+    sync::broadcast::Sender,
     time::{
         interval,
         Interval,
     },
 };
 
 pub use anyhow::Result;
+use tokio::sync::broadcast::Receiver;
+
+pub struct DaBlockCostsProvider<Source: DaBlockCostsSource + 'static> {
```

Comment:
Why do we want to wrap `DaBlockCostsService` into `DaBlockCostsProvider`? The gas price service already knows about `DaBlockCostsService` type. We can just work directly with this type.

The gas price service can subscribe in `new_service` from the shared state of the `DaBlockCostsService`

Reply:
my initial impl did this, it seemed cleaner to use the `DaBlockCostsProvider` to wrap over the service, and have only one field on the `GasPriceService`, which is responsible to get new block costs and the service handle :)

Reply:
I would prefer if we maintained this structure instead :)